tests/e2e/api/transaction_arguments.test.ts

```diff
@@ -0,0 +1,390 @@
+    });
+
+    accountResources.forEach((resource) => {
+      const data = resource.data as any;
+      if (data.empty_object_1 !== undefined) {
+        moduleObjects.push(new MoveObject(data.empty_object_1.inner));
+        moduleObjects.push(new MoveObject(data.empty_object_2.inner));
+        moduleObjects.push(new MoveObject(data.empty_object_3.inner));
+      }
+    });
+
+    // Transaction builder helpers
+    // single signer
+    rawTxnHelper = async (
+      sender: Account,
+      functionName: string,
+      typeArgs: TypeTag[],
+      args: Array<EntryFunctionArgumentTypes>,
+    ): Promise<UserTransactionResponse> => {
+      const rawTxn = await aptos.generateTransaction({
+        sender: sender.accountAddress.toString(),
+        data: {
+          function: `0x${account1.accountAddress.toStringWithoutPrefix()}::tx_args_module::${functionName}`,
+          type_arguments: typeArgs,
+          arguments: args,
+        },
+      });
+      const signedTxn = await aptos.signTransaction({
+        signer: sender,
+        transaction: rawTxn,
+      });
+      const txnHash = await aptos.submitTransaction({
+        transaction: rawTxn,
+        senderAuthenticator: signedTxn,
+      });
+      const response = await aptos.waitForTransaction({
+        txnHash: txnHash.hash,
+      });
+      return response as UserTransactionResponse;
+    };
+
+    // multi agent/fee payer
+    rawTxnMultiAgentHelper = async (
+      sender: Account,
+      functionName: string,
+      typeArgs: TypeTag[],
+      args: Array<EntryFunctionArgumentTypes>,
+      secondarySignerAddresses: AccountAddress[],
+      feePayerAddress?: AccountAddress,
+    ): Promise<UserTransactionResponse> => {
+      // TODO: Combine these to one type later. This should be possible but isn't right now due to typescript
+      // compile time errors because it can't resolve the types
+      const transaction = await (async () => {
+        if (secondarySignerAddresses.length == 0) {
+          const response = await aptos.generateTransaction({
+            sender: sender.accountAddress.toString(),
+            data: {
+              function: `0x${account1.accountAddress.toStringWithoutPrefix()}::tx_args_module::${functionName}`,
+              type_arguments: typeArgs,
+              arguments: args,
+            },
+            feePayerAddress: feePayerAddress!.data,
+          });
+          return response;
+        } else if (feePayerAddress === undefined) {
+          return await aptos.generateTransaction({
+            sender: sender.accountAddress.toString(),
+            data: {
+              function: `0x${account1.accountAddress.toStringWithoutPrefix()}::tx_args_module::${functionName}`,
+              type_arguments: typeArgs,
+              arguments: args,
+            },
+            secondarySignerAddresses: secondarySignerAddresses.map((address) => address.data),
+          });
+        } else {
+          return await aptos.generateTransaction({
```

Comment:
:/ super fragile, but we'll have to come back

Reply:
Yeah, the only reason I did it like this is because the types weren't resolving correctly because of the `?: undefined`, otherwise it should have correctly resolved to a general type I could use